src/test/mod.rs

```diff
@@ -448,6 +485,27 @@ impl TestEnvironment {
         })
     }
 
+    pub(crate) async fn async_db(&self) -> &TestDatabase {
+        let config = self.config().clone();
+        let runtime = self.runtime().clone();
+        let instance_metrics = self.instance_metrics().clone();
+        if self.db.get().is_none() {
```

Comment:
We could maybe switch `self.db` to a `tokio::sync::OnceCell`, and have `db()` just do `self.runtime().block_on(self.async_db())`.

Reply:
good idea! looks much nicer (apart from the remainin `spawn_blocking` ;) )